Author: Jasper Grimm <jasper.grimm@york.ac.uk>
Ensure that OpenEXR (python) can find the OpenEXR include and library files
diff -Nru OpenEXR-1.3.9.orig/setup.py OpenEXR-1.3.9/setup.py
--- OpenEXR-1.3.9.orig/setup.py	2024-02-27 14:20:58.796514598 +0000
+++ OpenEXR-1.3.9/setup.py	2024-02-27 14:29:13.274992396 +0000
@@ -2,7 +2,7 @@
 from distutils.extension import Extension
 from distutils.command.build_py import build_py as _build_py
 
-from os import system
+from os import path, environ
 import platform
 
 from distutils.core import setup, Extension
@@ -21,12 +21,7 @@
 """
 
 
-print("Looking for libOpenEXR...")
-if platform.system() == "Linux" and system("ldconfig -p | grep libOpenEXR"):
-    # There is no libOpenEXR, probably an old version of OpenEXR
-    libraries=['Iex', 'Half', 'Imath', 'IlmImf', 'z']
-else:
-    libraries=['Iex', 'OpenEXR', 'z']
+libraries=['Iex', 'OpenEXR', 'z']
 
 extra_compile_args = ['-g', '-DVERSION="%s"' % VERSION]
 if platform.system() == 'Darwin':
@@ -41,18 +36,12 @@
   description = "Python bindings for ILM's OpenEXR image file format",
   long_description = DESC,
   version=VERSION,
-  ext_modules=[ 
+  ext_modules=[
     Extension('OpenEXR',
               ['OpenEXR.cpp'],
-              include_dirs=['/usr/include/OpenEXR',
-                            '/usr/local/include/OpenEXR',
-                            '/opt/local/include/OpenEXR',
-                            '/usr/include/Imath',
-                            '/usr/local/include/Imath',
-                            '/opt/local/include/Imath'],
-              library_dirs=['/usr/lib',
-                            '/usr/local/lib',
-                            '/opt/local/lib'],
+              include_dirs=[path.join(environ['EBROOTOPENEXR'], 'include', 'OpenEXR'),
+                            path.join(environ['EBROOTIMATH'], 'include', 'Imath')],
+              library_dirs=[path.join(environ['EBROOTOPENEXR'], 'lib')],
               libraries=libraries,
               extra_compile_args=extra_compile_args)
   ],
